File: app/src/main/java/com/example/simonastojanovic/memeapp/view/MainActivityModule.java
package com.example.simonastojanovic.memeapp.view;

import android.arch.lifecycle.ViewModelProviders;

import com.example.simonastojanovic.memeapp.viewmodel.MemeViewModel;

import dagger.Module;
import dagger.Provides;

@Module
public class MainActivityModule {

    @Provides
    MemeViewModel provideMemeViewModel(MainActivity mainActivity) {
        return ViewModelProviders.of(mainActivity).get(MemeViewModel.class);
    }
}

File: app/src/main/java/com/example/simonastojanovic/memeapp/view/MainActivityModule.kt
package com.example.simonastojanovic.memeapp.view

import android.arch.lifecycle.ViewModelProviders
import android.content.Context

import com.example.simonastojanovic.memeapp.viewmodel.MemeViewModel

import dagger.Module
import dagger.Provides

@Module
class MainActivityModule {

    @Provides
    fun provideMemeViewModel(mainActivity: MainActivity): MemeViewModel {
        return ViewModelProviders.of(mainActivity).get(MemeViewModel::class.java)
    }

    @Provides
    fun provideMemeAdapter(context: Context): MemeAdapter {
        return MemeAdapter(context)
    }
}
